src/views/components/survey_components/question_time_setting.vue

```diff
@@ -0,0 +1,365 @@
+        // Fetch the list of questions from the API call
+        let fetched = await fetch(
+            "get_info.php?what=surveys&user_id=admin0&survey_id=1"
+        );
+        let fetchedJSON = await fetched.json();
+        // Parse the survey_package returned back into the data
+        if (fetchedJSON.TYPE === "error") {
+            // TODO: redirect to the 404 page
+            return 404;
+        } else if (fetchedJSON.TYPE === "survey_package") {
+            // assign value to the 'num_locked'
+            this.survey_package = fetchedJSON.DATA;
+            if (fetchedJSON.DATA.dept_number_locked != null) {
+                this.num_locked += parseInt(fetchedJSON.DATA.dept_number_locked);
+            }
+            if (fetchedJSON.DATA.course_number_locked != null) {
+                this.num_locked += parseInt(fetchedJSON.DATA.course_number_locked);
+            }
+            // Assign question package to this.questions
+            this.questions = fetchedJSON.DATA.questions;
+            // Sort the questions in the list to make it obey the order of positions
+            this.questions.sort(this.compare("position"));
+            // Determine the indices that should contain checkboxes
+            this.checkbox_indices.push(this.num_locked);
+            this.checkbox_indices.push(this.num_locked + 1);
+
+            // Set the default survey start and end time according to the JSON data
+            // sent back
+            let [date, time] = fetchedJSON.DATA.start_time.split(" ");
+            this.start_time = date + "T" + time;
+            [date, time] = fetchedJSON.DATA.end_time.split(" ");
+            this.end_time = date + "T" + time;
+        }
+        // Fill in the question_id_array
+        for (let i = 0; i < this.questions.length; i++) {
+            this.question_id_array.push(this.questions[i].question_id);
+        }
+        // Fetch all the questions back from the API
+        fetched = await fetch(
+            "get_info.php?what=questions"
+        );
+        fetchedJSON = await fetched.json();
+        // Determine which question objects to put into the backup_questions list
+        this.backup_questions = fetchedJSON.DATA.filter(this.question_filter);
+    },
+    methods: {
+        // helper function for filtering out the questions in the backup question list
+        question_filter: function (question) {
+            // return false if it is already in the question_id_array
+            return !(this.question_id_array.indexOf(question.question_id) >= 0);
+        },
+        // cancel the edit panel
+        cancel: function () {
+            this.selected = null;
+            this.edit_click = false;
+        },
+
+        // Checkbox on click action
+        checkbox_click: function (question) {
+            if (this.questions.indexOf(question) === this.checkbox_indices[0]) {
+                this.$set(this.check_status, 0, 1 - this.check_status[0]);
```

Comment:
By `1 - check_status` do you mean `!check_status` ?

Reply:
Sure, I will change this.